File: cpp/user_input.cpp
#include <iostream>

// cout << (insertion operator)
// cin >> (extraction operator)

int main()
{

	std::string name;
	int age;

	std::cout << "How old are you?: ";
	std::cin >> age;

	std::cout << "What's your full name?: ";
	// to accept a string with whitespaces
	/* std::getline(std::cin, name); */
	// Example #2 
	// 'std::ws' will eliminate new line character or any whitespaces
	// with 'std::cin' >> there will be a new line '\n' after
	std::getline(std::cin >> std::ws, name);


	std::cout << "Hello " << name << '\n';
	std::cout << "You are " << age << " years old." << '\n'; 

	return 0;
}

File: java/userInput.java
// Scanner will handle user inputs
import java.util.Scanner; // import this library when using Scanner 

public class userInput
{
	public static void main(String[] args)
	{
		Scanner scanner = new Scanner(System.in);	

		System.out.print("What is your name?: ");
		String name = scanner.nextLine();
		System.out.print("How old are you?: ");
		int age = scanner.nextInt();
		// use after using 'nextInt();' to clear the new line '\n' of the scanner
		scanner.nextLine();
		System.out.print("What is your favorite food?: ");
		// 'nextLine();' always clear the scanner after the input
		String food = scanner.nextLine();

		System.out.println("Hello " + name);
		System.out.println("You are " + age + " years old");
		System.out.println("You like " + food);
	}// end main
}// end class
